fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/logical/LogicalHudiScanOperator.java

```diff
@@ -3,40 +3,23 @@
 package com.starrocks.sql.optimizer.operator.logical;
 
 import com.google.common.base.Preconditions;
-import com.google.common.collect.Lists;
-import com.google.common.collect.Maps;
 import com.google.common.collect.Sets;
 import com.starrocks.catalog.Column;
 import com.starrocks.catalog.HudiTable;
-import com.starrocks.catalog.PartitionKey;
 import com.starrocks.catalog.Table;
 import com.starrocks.sql.optimizer.operator.OperatorType;
 import com.starrocks.sql.optimizer.operator.OperatorVisitor;
+import com.starrocks.sql.optimizer.operator.PredictsPushDown;
+import com.starrocks.sql.optimizer.operator.PredictsSummary;
 import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
 import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
 
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
-public class LogicalHudiScanOperator extends LogicalScanOperator {
+public class LogicalHudiScanOperator extends LogicalScanOperator implements PredictsPushDown {
```

Comment:
actually i suggest remove PredictsPushDown and add apis in LogicalScanOperator with default implementations throw not supported exception

Reply:
and we can rename PredictsSummary to ScanOperatorPredicates? and add HiveBasedScanOperatorPredicates implementation

Reply:
renamed.

I think interface is more flexible and compatible, but of course we can use default implementations to do the same thing. 
In this scenario do you have any guidelines or principles related to coding refactor?

Reply:
i left an other comment "I tend to remove this and add a base class LogicalHiveOperator and LogicalHudiOperator. Since this class name is a little misleading, we have many transformers and rule implementations for PushDown." @miomiocat